ethcore/src/engines/authority_round/mod.rs

```diff
@@ -806,19 +813,25 @@ impl AuthorityRound {
 			return;
 		}
 
-		if let (true, Some(me)) = (current_step > parent_step + 1, self.signer.read().as_ref().map(|s| s.address())) {
+		if let (true, me) = (current_step > parent_step + 1, self.address()) {
 			debug!(target: "engine", "Author {} built block with step gap. current step: {}, parent step: {}",
 				   header.author(), current_step, parent_step);
-			let mut reported = HashSet::new();
-			for step in parent_step + 1..current_step {
-				let skipped_primary = step_proposer(validators, header.parent_hash(), step);
-				// Do not report this signer.
-				if skipped_primary != me {
-					// Stop reporting once validators start repeating.
-					if !reported.insert(skipped_primary) { break; }
-					self.validators.report_benign(&skipped_primary, set_number, header.number());
- 				}
- 			}
+			if validators.contains(header.parent_hash(), &me) {
```

Comment:
Since now `me` can be `Address::zero` if signer is not configured I wonder if that wouldn't break some assumptions here if `Address:zero` is indeed a member of `validators` - while I know this shouldn't happen I think I kind of preferred the code when it was using `Option<>`